make/test.mk

```diff
@@ -26,3 +38,41 @@ test-with-coverage:
 upload-codecov-report: 
 	# Upload coverage to codecov.io
 	bash <(curl -s https://codecov.io/bash) -f $(COV_DIR)/coverage.txt -t 51cc45ad-2e54-4e68-98cb-8ab15cf2b8df
+
+############################################################
+#
+# End-to-end Tests
+#
+############################################################
+
+.PHONY: test-e2e
+## Runs the e2e tests locally
+test-e2e: e2e-setup 
+	$(info Running E2E test: $@)
+ifeq ($(OPENSHIFT_VERSION),3)
+	$(Q)oc login -u system:admin
+endif
+	$(Q)operator-sdk test local ./test/e2e --namespace $(TEST_NAMESPACE) --up-local --go-test-flags "-v -timeout=15m"
+
+e2e-setup: ./vendor get-test-namespace e2e-cleanup docker-image
+	$(Q)oc new-project $(TEST_NAMESPACE) --display-name e2e-tests
+	$(Q)-oc apply -f ./deploy/service_account.yaml 
+	$(Q)-oc apply -f ./deploy/role.yaml 
+	$(Q)sed -e "s,REPLACE_NAMESPACE,$(TEST_NAMESPACE)," ./deploy/role_binding.yaml | oc apply -f -
+	$(foreach crd_file,$(wildcard deploy/crds/*.yaml), \
+		oc apply -f $(crd_file); \
+	)
+	$(eval OPERATOR_IMAGE_NAME := $(shell minishift openshift registry)/${TEST_NAMESPACE}/${GO_PACKAGE_REPO_NAME}:${GIT_COMMIT_ID_SHORT})
```

Comment:
minishift? It's not going to work on openshift-ci, is it? Or I'm missing something?

Reply:
ah good point! I'll keep that for the local tests, and I'll see if we can have something similar with the test cluster or if we need to push the image to quay.io